### ISSUE TITLE
test: add convergence tests for Annealing, Metropolis, and Genetic

### ISSUE THREAD

**Issue by kovacoj** [kovacoj1/optimizers#61]
## Problem

`test_progress.py` has convergence tests for Newton, LM, EKF, and KF (verifying that loss decreases), but there are **no convergence tests** for Annealing, Metropolis, or Genetic. The runtime tests only verify that `step()` returns without error and the return type is correct.

These stochastic optimizers should be tested on a simple problem where they can reasonably be expected to make progress over enough steps. Even though individual steps may not always decrease loss, the trend over many steps should.

## Proposal

Add tests similar to the existing `test_newton_reduces_quadratic_loss` pattern:

```python
def test_annealing_reduces_quadratic_loss_over_many_steps():
    x = _vector_param()
    optimizer = Annealing([x])
    # Run enough steps with a reasonable temperature
    ...

def test_metropolis_reduces_quadratic_loss_over_many_steps():
    ...

def test_genetic_reduces_quadratic_loss():
    # Genetic already uses Adam internally, should converge on simple problems
    ...
```

Since these are stochastic, use a fixed seed and a generous step count. The assertion should be that loss is strictly lower after N steps, not that it decreases monotonically.

**Comment by kovacoj** [comment]
## Proposed fix

Add convergence tests for the three stochastic optimizers. Since individual steps may not decrease loss, use fixed seeds and enough steps to see a clear trend:

```python
import torch
from optimizers import Annealing, Metropolis, Genetic


def _vector_param(values=(1.5, -1.0)):
    return torch.nn.Parameter(torch.tensor(values))


def _quadratic_loss(x):
    return ((x - torch.zeros(2)) ** 2).sum()


def test_annealing_reduces_quadratic_loss_over_many_steps():
    torch.manual_seed(42)
    x = _vector_param()
    optimizer = Annealing([x])
    with torch.no_grad():
        before = _quadratic_loss(x).item()
    for _ in range(200):
        optimizer.step(lambda: _quadratic_loss(x))
    with torch.no_grad():
        after = _quadratic_loss(x).item()
    assert after < before


def test_metropolis_reduces_quadratic_loss_over_many_steps():
    torch.manual_seed(42)
    x = _vector_param()
    optimizer = Metropolis([x])
    with torch.no_grad():
        before = _quadratic_loss(x).item()
    for _ in range(200):
        optimizer.step(lambda: _quadratic_loss(x))
    with torch.no_grad():
        after = _quadratic_loss(x).item()
    assert after < before


def test_genetic_reduces_quadratic_loss():
    x = _vector_param()
    optimizer = Genetic([x])
    optimizer.pop_size = 8
    optimizer.population = optimizer.params.unsqueeze(0).repeat(optimizer.pop_size, 1)
    with torch.no_grad():
        before = _quadratic_loss(x).item()
    for _ in range(5):
        optimizer.step(lambda: _quadratic_loss(x))
    with torch.no_grad():
        after = _quadratic_loss(x).item()
    assert after < before
```

**Design choices:**
- Annealing/Metropolis: 200 steps with seed 42 is generous enough that the trend overwhelms the stochastic noise.
- Genetic: Uses small population (8) since the test needs to be fast; 5 generations is enough for the internal Adam helper + selection to make progress on a simple quadratic.
- The assertion is strict (`after < before`), not monotonic — we only check the endpoint, not every intermediate step.
